src/m365/yammer/commands/yammer-search.ts

```diff
@@ -0,0 +1,214 @@
+          const resultTopics = results.topics;
+          const resultGroups = results.groups;
+          const resultUsers = results.users;
+
+          let continueprocessing = true;
+          let resultsFound = false;
+
+          if (resultMessages.length > 0) {
+            resultsFound = true;
+            this.messages = this.messages.concat(resultMessages);
+
+            if (args.options.limit && this.messages.length > args.options.limit) {
+              continueprocessing = false;
+              this.messages = this.messages.slice(0, args.options.limit);
+            }
+          }
+
+          if (resultTopics.length > 0) {
+            resultsFound = true;
+            this.topics = this.topics.concat(resultTopics);
+
+            if (args.options.limit && this.topics.length > args.options.limit) {
+              continueprocessing = false;
+              this.topics = this.topics.slice(0, args.options.limit);
+            }
+          }
+
+          if (resultGroups.length > 0) {
+            resultsFound = true;
+            this.groups = this.groups.concat(resultGroups);
+
+            if (args.options.limit && this.groups.length > args.options.limit) {
+              continueprocessing = false;
+              this.groups = this.groups.slice(0, args.options.limit);
+            }
+          }
+
+          if (resultUsers.length > 0) {
+            resultsFound = true;
+            this.users = this.users.concat(resultUsers);
+
+            if (args.options.limit && this.users.length > args.options.limit) {
+              continueprocessing = false;
+              this.users = this.users.slice(0, args.options.limit);
+            }
+          }
+
+          // does not process more queries if we are not able to return a complex object 
+          if (resultsFound && continueprocessing && args.options.output === 'json') {
+            this
+                .getAllItems(logger, args, ++page)
+                .then((): void => {
+                  resolve();
+                }, (err: any): void => {
+                  reject(err);
+                });
+          }
+          else {
+            resolve();
+          }
+        }, (err: any): void => {
+          reject(err);
+        });
+    });
+  };
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this.summary = {
+      messages: 0,
+      groups: 0,
+      topics: 0,
+      users: 0
+    }
+    this.messages = [];
+    this.groups = [];
+    this.topics = [];
+    this.users = [];
+
+    this
+      .getAllItems(logger, args, 1)
+      .then((): void => {
+        if (args.options.output === 'json') {
+          logger.log(
+            {
+              summary: this.summary,
+              messages: this.messages,
+              users: this.users,
+              topics: this.topics,
+              groups: this.groups,
+            });
+        }
+        else {
+          logger.log(this.summary);
```

Comment:
If I'm searching, I'd expect to see the results rather than stats. Is there a way to provide a simple overview of results matching the query, eg. title + URL, something that would be useful by itself?

Reply:
The challenge I see here is that we have different types of objects that are returned.

Messages, Topics, Groups, and Users.

The only common denominator I see is the ID and DisplayName. Everything else is somewhat custom. We could think of providing multiple tabular responses (do we have already something like that?) or to return following format:

ID, DisplayName, Type
where type is "GROUP, USER, TOPIC, MESSAGE"

Alternatively we could limit the search results to a type and return the whole object by passing a type parameter like "MESSAGE,TOPIC,USER,GROUP,SUMMARY" and then return the results. I would provide this option just for the "TEXT" output.

Reply:
Is there a URL for each of these objects to open them in context of Yammer? What is the context of search? What would you want to do with the results next? I think answering these questions, will help us decide what kind of information we should display.

Reply:
There is just one endpoint that allows to perform a text search. It will then return a summary and all the objects found in that search. You do not have many other filtering capabilities. The added filtering capabilities and data output capabilities are then performed client side. 

I am still thinking that the JSON output is more valuable in terms of automation. I do not think that I would use the CLI to perform a basic text search. 

The only reasonable solution might be to provide some "enhanced" filtering capabilities with added parameters and filter with "MESSAGE,TOPIC,USER,GROUP,SUMMARY" the contents client side. If somebody wants to have everything in one shot, then JSON should be used.

Reply:
I didn't mean the ability to filter, but rather having a link to the result, which you could open in the browser or include in a report. So when you execute the command, you get a table of title, type and URL. If you want all data for automation, you can grab JSON.